### PR TITLE
fixing add_pass_through_columns macro

### PR DESCRIPTION
**Pull request by jamesrayoub**
**What change does this PR introduce?** 
For instances where the source table does not include the additional pass through columns indicated in the project.yml file, the `fill_staging_columns()` macro will fail, specifically line 6-12: 
```
{% if column.name|lower in source_column_names -%}
    {{ fivetran_utils.quote_column(column) }} as 
    {%- if 'alias' in column %} {{ column.alias }} {% else %} {{ fivetran_utils.quote_column(column) }} {%- endif -%}
{%- else -%}
    cast(null as {{ column.datatype }})
    {%- if 'alias' in column %} as {{ column.alias }} {% else %} as {{ fivetran_utils.quote_column(column) }} {% endif -%}
{%- endif -%} 
```

This is because the pass through columns do not specify datatype and will result in compiled sql like `cast(null as )` which will fail. Modification of the `add_pass_through_columns` macro will allow for the addition of a datatype key in the project.yml  or use string as the default.

**If this PR introduces a modification to an existing macro, which packages is the macro currently present in and what steps were taken to test compatibility across packages?** 
* tested in dbt_hubspot_source

**Did you update the README to reflect the macro addition/modifications?** 

- [ ] Yes
- [x] No (provide further explanation)

### CODE REVIEW

**Comment by jamesrayoub** [comment]
@fivetran-joemarkiewicz here is my suggested edit

**Comment by jamesrayoub** [comment]
The reason for this PR is due to the fact that I'm trying to replicate the production models in a staging environment using staging instances of our tools which don't always have all of the same custom fields as in production; i.e. not every new custom field was tested in staging before creating in prod.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Thanks for opening this @jamesrayoub and the reasoning you provided makes sense.

I tested this and it looks good on my end! I do need to make a few minor edits before and merge to the appropriate branch prior to release. Therefore, I will merge to a staging feature branch and will merge to the release branch after. I will tag you in the following PR so you can be up to date.

**Pull request merged**
